packages/core/react/src/WalletProvider.tsx

```diff
@@ -87,14 +89,7 @@ export function WalletProvider({
     const changeWallet = useCallback(
         (nextWalletName: WalletName<string> | null) => {
             if (walletName === nextWalletName) return;
-            if (
-                adapter &&
-                // Selecting a wallet other than the mobile wallet adapter is not
-                // sufficient reason to call `disconnect` on the mobile wallet adapter.
-                // Calling `disconnect` on the mobile wallet adapter causes the entire
-                // authorization store to be wiped.
-                adapter.name !== SolanaMobileWalletAdapterWalletName
```

Comment:
Can you explain to me the purpose of this change?

Reply:
Oh this was removed with the assumption that we needed `disconnect` on the MWA to be called, so that it would deauthorize the MWA token (if it were to be added)

given that this is no longer the case I'll revert this